### ISSUE TITLE
de_dot plugin

### ISSUE THREAD

**Issue by dvenza** [docker-library/logstash#45]
Please include the de_dot plugin in the image by default.

**Comment by holmjoel** [comment]
I can't figure out how to add plugins not included in the image. An explanation of that in the read me could probably help with this request and similar in the future.

I'm trying to use the https://github.com/revpoint/logstash-codec-avro_schema_registry and can't figure out how. Anyone that could point me in the right direction?

**Comment by mi-hol** [comment]
@tianon with 5 thumbs up, would this qualify as a clear vote for including the plugin by default?
Maybe the community would benefit from knowing the rules for considering a request. Other projects publish these rules via links in their readme.md

**Comment by yosifkit** [comment]
I think our current ruling on this image is to only include what is recommended by upstream.  From the [elastic.co logstash guide](https://www.elastic.co/guide/en/logstash/current/plugins-filters-de_dot.html) `de_dot` section:

> This does not ship with Logstash by default, but it is easy to install by running `bin/logstash-plugin install logstash-filter-de_dot`.

So, to translate that to a Dockerfile, it should be as simple as this:

```dockerfile
FROM logstash:5
RUN logstash-plugin install logstash-filter-de_dot
```

I'll add something to the Docker Hub description about installing plugins. (https://github.com/docker-library/docs/pull/771)

**Comment by mi-hol** [comment]
Thanks @yosifkit this rule makes perfect sense to me. 
Best approach for dealing with the request would seem to ask requestor to address the need to upstream and close it here.

**Comment by yosifkit** [comment]
Thanks @mi-hol.

Closing this issue in favor of added documentation.  If you desire this plugin or any other to be installed by default, please file an issue with logstash upstream: https://github.com/elastic/logstash/issues.

**Comment by blubbi321** [comment]
de_dot does not seem to be included in 5.5.0, even though
https://www.elastic.co/guide/en/logstash/5.5/plugins-filters-de_dot.html
now claims that it is included (warning box from <= 5.4 is missing)

update: ok, apparently the plugin is also not included in the ls 5.5 distribution when you download the binaries directly ..
